### PR TITLE
feat(nmt): Add S2S ST rpc's

### PR DESCRIPTION
**Pull request by junkin**
feat: add speech to text, speech to speech streaming API's

### CODE REVIEW

File: riva/proto/riva_nmt.proto [hunk 1]
```diff
@@ -21,12 +24,88 @@ service RivaTranslation {
...
 
+/*
+* Configuration for Translate S2S.  reuse existing protos from other services.
+*/
+message StreamingTranslateSpeechToSpeechConfig {
+   nvidia.riva.asr.StreamingRecognitionConfig asrConfig = 1;  //from riva_asr.proto
+   SynthesizeSpeechConfig ttsConfig = 2; //from riva_tts.proto
+   string source_language_code = 3;
+   string target_language_code = 4;
+   AudioEncoding t = 5;
```
Comment (messiaen) [inline riva/proto/riva_nmt.proto]: AudioEncoding is part of both asr config and tts config.  Why have a separate parameter here?  I guess I could ask the same question for source_language_code.

File: riva/proto/riva_nmt.proto [hunk 2]
```diff
@@ -21,12 +24,88 @@ service RivaTranslation {
...
+}
+
+message StreamingTranslateSpeechToTextResponse {
+
+   repeated nvidia.riva.asr.StreamingRecognitionResult results = 1; //from riva_asr.proto
+}
+
+message StreamingTranslateSpeechToTextConfig {
+   nvidia.riva.asr.StreamingRecognitionConfig config = 1; //existing ASR config
+   bool interim_results = 2;  // not always supported
```
Comment (messiaen) [inline riva/proto/riva_nmt.proto]: Is this different than `interim_results` in asr streaming config?

File: riva/proto/riva_nmt.proto [hunk 3]
```diff
@@ -21,12 +24,88 @@ service RivaTranslation {
...
+*
+*/
+message StreamingTranslateSpeechToSpeechRequest {
+   oneof streaming_request {
+       StreamingTranslateSpeechToSpeechConfig config = 1;
+       bytes audio_content = 2;
+   }
+}
+
+message SynthesizeSpeechConfig {
```
Comment (messiaen) [inline riva/proto/riva_nmt.proto:69]: This should probably be moved to riva_tts.proto.  And TTS refactored to accommodate the config.

File: riva/proto/riva_nmt.proto [hunk 4]
```diff
@@ -21,12 +24,88 @@ service RivaTranslation {
...
+  rpc TranslateText(TranslateTextRequest) returns (TranslateTextResponse) {}
 
   // Lists the available language pairs and models names to be used for TranslateText
   rpc ListSupportedLanguagePairs(AvailableLanguageRequest) returns (AvailableLanguageResponse) {}
+
+  //streaming speech to text translation api.
+  rpc StreamingTranslateSpeechToText(stream StreamingTranslateSpeechToTextRequest)
+        returns (stream StreamingTranslateSpeechToTextResponse) {}
+
+  //offline speech to text translation api.
```
Comment (messiaen) [inline riva/proto/riva_nmt.proto]: remove commented out rpcs for now.

**Review by messiaen — CHANGES_REQUESTED** [review]

File: riva/proto/riva_nmt.proto (hunk 3 above)
Comment (junkin) [reply riva/proto/riva_nmt.proto:69]: TTS needs a revisit, out of scope for this change.  TTS operates per request and doesnt have a streaming concept,  This config defines how we call multiple TTS requests within a S2S stream.

File: riva/proto/riva_nmt.proto (hunk 1 above)
Comment (junkin) [reply riva/proto/riva_nmt.proto]: source/target lang are the NMT parameters.  They both could be infered from their respective ASR and TTS configs.
AudioEncoding will be removed.

File: riva/proto/riva_nmt.proto (hunk 2 above)
Comment (junkin) [reply riva/proto/riva_nmt.proto]: this is intended to allow the interim ASR transcription be returned alongside the translated text, will remove for now

File: riva/proto/riva_nmt.proto (hunk 1 above)
Comment (junkin) [reply riva/proto/riva_nmt.proto]: added nmt config message, removed audioencoding.

File: riva/proto/riva_nmt.proto (hunk 4 above)
Comment (junkin) [reply riva/proto/riva_nmt.proto]: done

**Review by messiaen — APPROVED** [review]

**Pull request merged**